virttest/utils_libvirt/libvirt_network.py

```diff
@@ -346,10 +346,29 @@ def change_tcp_config(expected_dict):
 
 def check_listen_address(params):
     """
-    Check listen address by ss command
+    Check listen address and port by ss command
 
     :param params: dictionary with the test parameter, get listen address
     """
-    listen_address = params.get("listen_address")
-
-    process.run(f"ss -tnap|grep qemu-kvm| grep '{listen_address}'", shell=True, ignore_status=False)
+    def _check_str_in_result(string, result):
+        """
+        Check string in result
+
+        :param string: the string to check
+        :param result: check string in result
+        """
+        check_list = eval(string)
+        for value in check_list:
+            ret = re.search(value, result)
+            if not ret:
+                raise exceptions.TestFail("Pattern '%s' is not matched in '%s'" % (value, result))
+
+    check_listening_on = params.get("check_listening_on")
+    check_established_conn = params.get("check_established_conn")
+
+    cmd = "ss -tnap|grep qemu-kvm"
+    result = remote.run_remote_cmd(cmd, params).stdout_text.strip()
+    if check_listening_on:
+        _check_str_in_result(check_listening_on, result)
+    if check_established_conn:
+        _check_str_in_result(check_established_conn, result)
```

Comment:
Can we just return cmd object in L370 and check the result via libvirt.check_result or libvirt.check_status_output or others in vt?

Reply:
Updated to libvirt.check_status_output, thanks.